query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -348,6 +405,8 @@ private String convertLiteralToString(LiteralConstant literal, Params.Builder pa
         paramsBuilder.addDoubleParam(value.getDouble());
         break;
       case BYTES:
+        ret = QUESTION_MARK;
+        paramsBuilder.addBytesStringParam(Hex.encodeHexString(value.getBytes().toByteArray()));
```

Comment:
Careful with `Hex.encodeHexString()` especially if you need to access the ByteBuffer later on.(Which you do not need to over here)

Reply:
Also what happens when this is `'null'`

Reply:
Actually, would this ever even happen? None of our attributes are typed as `BYTES` right?

Reply:
yes. we don't have BYTES exposed at the gateway layer. The default value for bytes in proto is `bytes[0]`. So, this will be fine if we ever expose it.